src/test/java/org/fcrepo/client/FcrepoResponseTest.java

```diff
@@ -69,4 +79,103 @@ public void testResponse() throws IOException {
                     "<http://example.org/book/3> <dc:title> \"Title\" .");
     }
 
+    /**
+     * Demonstrates that response objects are <em>not</em> {@code close()}ed by default, that the state of
+     * {@link FcrepoResponse#closed} is set appropriately when {@link FcrepoResponse#close()} is invoked under normal
+     * (i.e. no exception thrown during {@code close()}) conditions, and that {@link InputStream#close()} is not invoked
+     * repeatedly after the {@code FcrepoResponse} has been {@code close()}ed.
+     *
+     * @throws Exception if something exceptional happens
+     */
+    @Test
+    public void testClosableReleasesResources() throws Exception {
+        final InputStream mockBody = mock(InputStream.class);
+        final FcrepoResponse underTest = new FcrepoResponse(
+                URI.create("http://localhost/foo"), 201, "text/plain", URI.create("http://localhost/bar"), mockBody);
+
+        assertFalse("FcrepoResponse objects should not be closed until close() is invoked.", underTest.isClosed());
+
+        underTest.close();
+        assertTrue(underTest.isClosed());
+        verify(mockBody, times(1)).close();
+
+        underTest.close();
+        assertTrue(underTest.isClosed());
+        verify(mockBody, times(1)).close();
+    }
+
+    /**
+     * Demonstrates that if an {@code IOException} is thrown by {@link FcrepoResponse#close()}, <em>and</em> an
+     * exception is thrown inside of a client's {@code try} block, the {@code IOException} from the {@code close()}
+     * method is properly appended as a suppressed exception.
+     *
+     * @throws Exception if something exceptional happens
```

Comment:
This is pretty much why I prefer tests to `throw` something more specific than `Exception`.